test/rpc_test.py

```diff
@@ -858,3 +859,13 @@ def test_func():
 
         if TEST_CONFIG.rpc_backend == RpcBackend.PROCESS_GROUP:
             self.assertEqual(test_func(), "expected result")
+
+    def test_process_group_agent_destructor(self):
+        dist.init_process_group(backend="gloo", init_method=self.init_method, timeout=timedelta(seconds=10))
```

Comment:
Can you enhance with tests by doing some RPCs as well? I think that might trigger some of the issues with condition variables and mutexes.